### PR TITLE
GitHub Deployment - Allow Automerge configuration

### PR DESCRIPTION
**Pull request by mikesplain**
Hi there, thanks for adding support for Deployments to the Notifications Engine.  I was testing out the support and ran into issues trying to deploy an older ref from the default branch.  The notifications controller would report an error when merging the default branch into the commit referenced.  Upon reading the docs, it looks like this is the [intended behavior](https://docs.github.com/en/rest/deployments/deployments?apiVersion=2022-11-28#create-a-deployment) so it seems reasonable to allow users to disable this feature when they would not like their deployment updated against the default branch. 

This PR was tested against an internal custom ArgoCD build and it resolved our issues with deployments only functioning on the latest commit in a default branch.  

I'm happy to make any required changes, just let me know.  Thanks!

### CODE REVIEW

**Comment by mikesplain** [comment]
Please hold off on this at the moment, I want to do another round of testing now that I got the tests and linter to pass.

**Comment by pasha-codefresh** [comment]
@mikesplain is it ready for review?

**Comment by mikesplain** [comment]
@pasha-codefresh It is, thank you! Just finished a second round of testing.

File: docs/services/github.md [hunk 1]
```diff
@@ -75,8 +75,11 @@ template.app-deployed: |
...
       logURL: "{{.context.argocdUrl}}/applications/{{.app.metadata.name}}?operation=true"
       requiredContexts: []
+      autoMerge: true
 ```
 
 **Notes**:
 - If the message is set to 140 characters or more, it will be truncated.
 - If `github.repoURLPath` and `github.revisionPath` are same as above, they can be omitted.
+- By default, github deployments utilize automerge to ensure the requested ref is up to date with the default branch.
+  Setting this option to `false` is required if you would like to deploy older refs in your default branch.
```
Comment (pasha-codefresh) [inline docs/services/github.md:85]: Can we also mention that this parameter is optional and by default it is false, i would write opposite , like: Setting this option to `true` is required ....

**Review by pasha-codefresh — CHANGES_REQUESTED** [review]

**Comment by pasha-codefresh** [comment]
@mikesplain LGTM, just small comment about documentation

**Comment by carloscastrojumo** [comment]
I was just looking for this feature @mikesplain . Thanks for the PR. Are you still available to address the issues requested by the reviewers?

**Comment by mikesplain** [comment]
Yes, found a bug and had to fix the code slightly.  Tested it internally and it works fine just need to rebase. I'll update the PR tomorrow ASAP.

File: docs/services/github.md (hunk 1 above)
Comment (mikesplain) [reply docs/services/github.md:85]: Sorry for the delay I realized that you're right, this PR set the value to false by default however the upstream API is true by default.  I had to modify the PR and language in the doc to reverse the behavior. I hope the language in the docs is more clear now. Thanks!

**Comment by mikesplain** [comment]
@pasha-codefresh I apologize for the long delay, life gets in the way. This is ready for review again and has been rebased against master!

After your comments I realized the PR worked properly if you set the new configuration however the default was false, rather than the Github default (and argocd notifications default, true), see https://docs.github.com/en/rest/deployments/deployments?apiVersion=2022-11-28#create-a-deployment for more details.  

I switched the use of pointers so we can check if the user has set the configuration and default it accordingly. I'm not sure if the project has a better way to handle this, I'm happy to amend with any suggestions. Thanks again

**Comment by carloscastrojumo** [comment]
Any news on this @pasha-codefresh ?

**Comment by pasha-codefresh** [comment]
Sorry, hadnt time to review, @carloscastrojumo @mikesplain , i will do it today / tomorrow

**Comment by pasha-codefresh** [comment]
LGTM, thank you

**Comment by mikesplain** [comment]
Great thanks so much @pasha-codefresh!

**Comment by pasha-codefresh** [comment]
Thank you for great job :) you can open PR to argo-cd with notification-engine update, you can send it here, i will help with review

**Pull request merged**